### DISCUSSION TITLE
Course Load & Combinations

### DISCUSSION THREAD

**Discussion by deepak0797** [snigi-gupta/AttendingUniversityAtBuffalo#51] [Q&A] [answered]
Hi Snigdha,
Thank you for creating this repository this is really great and helpful!!
I have shortlisted few courses but I haven't decided on the combinations because I'm not sure about the difficulty level and the best options.
I'm planning to specialize in Software Systems:

These are the courses I'm interested 
1.CSE 535-Information Retrieval by Rohini K. Srihari 
2.CSE-521-Operating Systems by Farshad Ghanei 
3.CSE-574-Introduction to Machine Learning offered by  Sargur Srihari & Mingchen Gao
4.CSE-586-Distributed Systems by Bina Ramamoorthy
5.CSE-611-Ms Project by Alan Hunt
6.CSE-560 Data Models and Query Languages by Sreyasee Das Bhattacharjee
7.CSE 531-Analysis of Algorithm I by Xin He

Below are my options;

Option-1:
IR, OS, Intro to ML, Distributed Systems

Option-2:
IR, OS, Intro to ML,611 by Alan hunt

Option-3:
IR, OS, Intro to ML, DMQL by Sreyasree

Option-4: (this I will consider if Kosar is going to take OS in Spring 22)
IR, Intro to ML, DMQL,611 by Alan Hunt
Option 5: ( I have included the Algo course here, thinking this might help me for preparing for Internships. but I heard that the Analysis of Algorithm course in the Spring Semester by Shi Le  is the best one for ALGO !!!.)
IR, OS, Analysis of Algorithms by Xin he, Intro To ML

Q1:Could you help me in ordering these options from very heavy course load to medium, easy course load, considering Part-time jobs and also considering this will be my very first semester and I'm a non-CS student with 3 years of experience in the Software Industry
Q2:What is your view on  OS and IR in the same semester? Is it advisable to take both in the same semester?
Q3:I know OS by Kosar will be the best, but I heard Kosar is not taking this course in recent years is that the case? I'm from non-CS so I personally wanted to understand how OS/memory allocation works so looking for the best.
Q4:Who is better  Sargur Srihari vs Mingchen Gao for Intro to ML, I know Varun is the BEST !! but he is not taking this fall:(
Q5:Comparison between Shi Le & Xin he for ALgo ?
Q6: What is your perspective on the 611 MS Project for Alan hunt, is it advisable to take it in the first semester?
Q7:Please provide your review about DMQL by Sreyasree

**Comment by snigi-gupta** [comment, marked as answer]
Hi @HariDeepak07,

IR  - Medium,
OS - Medium-Hard,
ML by Srihari - Easy
ML by Gao - Medium
Algo by Roger/Shi Li - Medium
611 by Hunt - Easy-Medium
DS by Bina - expecting it to be Medium (can't be sure since she is taking this course for the first time that I know off. Earlier Steve ko used to take it and it was Hard)
DMQL by Sryeashee (not sure, assuming it Medium if not Easy)

I don't think so I can rank these combinations for you since this is very subjective. For me OS is hard, but for someone who has worked in C before, OS is medium. So you see, it all depends on the individual opting for the course. I have listed the general connotation of the difficulty level of these courses for you. I would recommend you to refer the [combination list mentioned here](https://github.com/snigi-gupta/AttendingUniversityAtBuffalo/blob/main/Computer%20Science%20Important%20Tips/CourseReview.md) and select a combination.

I would recommend you to keep it a little light the first sem, since you will be going through a big change when moving countries. Cooking, doing household chores, etc will also take up time, and time management is a skill you will learn if you already are not good at it.

Don't bother with doing a part-time job. I always recommend students focus on their studies and getting an internship. Working at CDS will not do as much good, as hunting for internships/self-studying will do. Out of the courses you mentioned I personally took the following and here are my byte-sized inputs for you.

IR - I would recommend this course. Check out my IR repo to get an idea of the type of projects Srihari offers. Her courses are well structured, projects are resume-worthy, exams are tricky (for me it was, not a big fan of exams or MCQs) and I enjoyed learning about 'Life of a query' - something that is taught on Day 1 at Google. My grade was a B+, I missed out on an A- by 1 or 2 points. Still no regrets about taking this course. I used many instances/examples during this course in my interviews. Helped me a lot!

Algos by Roger - Algos was hard for me. My fundamentals were shaken before I arrived here and no one guided me to study DSA before I come here. My learning happened during this course. I did pretty bad in the 1st mid-term, and then gradually improved. It took some effort for me to be able to get a decent B. Understanding Roger's accent was a little difficult, but his slides are good. I still refer them. Algo is a subject that is not much prof dependent. There were many assignments and exams, so there were many chances to fix your scores. With Shi Li, I too have heard good things. I don't think so there will be much hard if you take either of the algos. Ultimately, you will have to spend time understanding the course.

I also took ML by Srihari - I won't blame the professor here, but how he has organized the course. There is too much to cover, and his teaching style is erratic, so I did not enjoy this class much. I did get an A though, but that is because the projects are very easy once you figure out what is that you are doing.  Plenty of medium articles to help you out with those projects.

611 by Hunt - I did not take this course, but I wanted to so I know a great deal about this one too. Great course to complete MS project requirement and add a project on your resume. I would recommend this course. You interact with real clients and develop a product. Usually ios apps, android apps or web development projects. This course is not difficult, just time taking for a couple of reasons. a) you work in a group b) weekly deliverables c) agile working so you incorporate changes/updates with the flow.

Since you have SW experience, 611 is not a compulsion for you. However, 611 is a course offered in all sems, so if for some reason you are tied up and cannot take it this sem, you can take it in the next sem too which is what usually most people do.

OS by Farshad - I have heard good things about this prof. I think he used to take Robotics too. He works really hard for his students, so if you are REALLY interested in OS, take and work hard. If Kosar is not teaching OS, never mind, you can always learn so much more by putting effort from your end.

DS and DMQL - I personally wouldn't recommend, but don't just go with what I have to say, go and audit the course. Not just this but others also. 1st week is for this only and see for yourself whether the prof's teaching style, course structure, and methods work for you or not.

Don't follow the herd. Pick courses that either interest you or will help you add a project to your resume or will help you get a job.

Also, if due to some constraint you are able to take a course only in a particular sem (for example Shi Li), then plan your courses for all three sems at once. Do broad planning. This hack helped me a lot since I knew which course to opt when and also had backup courses if for some reason a course was discontinued or canceled.

I hope this helps. If you have more questions, feel free to post there here. I recently started something to help out more people. If possible please [show your support here and encourage your friends too.](https://www.instagram.com/yourtechdiary/)

**Reply by deepak0797** [reply]
Wow! Thank you for answering and explaining everything !! This is really helpful . Once again Thank you so much !!

**Reply by snigi-gupta** [reply]
Just clarifying...DS and DMQL by these prof is not very highly suggested (again reiterating, do Audit to see if their teaching style works for you!). 
However I would suggest to take DS by Ethan Blanton if possible and learn how big tech companies are designing large scalable systems. Basically DS as a course is very useful for system design rounds.

If possible follow me on Instagram [yourtechdiary](https://www.instagram.com/yourtechdiary/), I am sharing useful resume tips that can help you in your internship search. :)

**Reply by deepak0797** [reply]
sure I'm planning to audit the courses and decide !! Once again thank you so much !!
